### PR TITLE
parse media item from within the ImageField itself

### PR DESCRIPTION
**Pull request by jeffsee55**
Previously the Cloudinary image parsing happened from a `parse` hook we were hijacking from the `useGraphqlForms` API. This would make it more "core", though I'm admittedly a little lost on what `media` vs `media.store` represents. And we don't have an optional type on the `parse` function from the `MediaStore` interface. Didn't want to add that until I get some more info on how this might be problematic.

EDIT: select "Ignore Whitespace" to see actual diff

### CODE REVIEW

File: packages/@tinacms/toolkit/src/packages/fields/plugins/ImageFieldPlugin.tsx [hunk 1]
```diff
@@ -68,62 +68,69 @@ export function usePreviewSrc(
...
+  const uploadDir = props.field.uploadDir || (() => '')
+
+  return (
+    <ImageUpload
+      value={value}
+      previewSrc={src}
+      loading={srcIsLoading}
+      onClick={() => {
+        const directory = uploadDir(props.form.getState().values)
+        cms.media.open({
```
Comment (Enigmatical) [inline packages/@tinacms/toolkit/src/packages/fields/plugins/ImageFieldPlugin.tsx:112]: Definitely see what you mean about `media` vs. `media.store`.  Are the APIs available to `media` more global?  As in, the act of `open` is universal, regardless of the `store`?

File: packages/@tinacms/toolkit/src/packages/fields/plugins/ImageFieldPlugin.tsx [hunk 2]
```diff
@@ -68,62 +68,69 @@ export function usePreviewSrc(
...
+    if (media) {
+      const mediaParse = cms.media.store
+
+      const parsedValue =
+        // @ts-ignore
+        typeof cms?.media?.store?.parse === 'function'
+          ? // @ts-ignore
+            cms.media.store.parse(media)
+          : // @ts-ignore
+            mediaParse.parse(media)
```
Comment (Enigmatical) [inline packages/@tinacms/toolkit/src/packages/fields/plugins/ImageFieldPlugin.tsx:97]: Trying to make sense of this... if `cms.media.store.parse` is a `function`, then use it.  If not... use `cms.media.store.parse`?  Might be missing the point here, but this almost reads like we are sourcing the same `parse()` method.

**Review by Enigmatical — COMMENTED** [review]
Not as familiar with the Cloudinary approach, but I left some comments.

File: packages/@tinacms/toolkit/src/packages/fields/plugins/ImageFieldPlugin.tsx (hunk 1 above)
Comment (jeffsee55) [reply packages/@tinacms/toolkit/src/packages/fields/plugins/ImageFieldPlugin.tsx:112]: This wasn't a change I made, just a whitespace diff, I probably mixed up the prettier formatter during migration

File: packages/@tinacms/toolkit/src/packages/fields/plugins/ImageFieldPlugin.tsx (hunk 2 above)
Comment (jeffsee55) [reply packages/@tinacms/toolkit/src/packages/fields/plugins/ImageFieldPlugin.tsx:97]: That's a mistake, I meant to just pass `media` for the fallback condition. Will fix

**Review by Enigmatical — APPROVED** [review]
I'm fine with this.  I still wouldn't mind having a discussion on `media` vs. `media.store`, but that doesn't have to happen with this field.

**Comment by jbevis** [comment]
Been reviewing this and revisiting our ```Media``` and ```MediaStore``` interfaces just to try and solidify what we're trying to do.

as far as ```media``` vs ```media.store``` is concerned, it would appear that the `parse` would be tied to the store instance and specifically how we're changing the data for a given Media file. The `media` key on the cms appears to be a way to scope the functionality and logic of the store under media, while also providing a developer the ability to control access to the store outside of the baked in screen plugin.

I'm a bit confused on where the `parse` function even comes from. @jeffsee55 you mentioned above how the MediaStore interface doesn't allow for an optional parse, but I don't even see where the MediaStore provides for any parse. Could just be I'm looking at outdated code though.

Overall, I agree with Chris, I think this is a fine solution for now. It does speak to the broader issue of maybe do a more wholesale reexamination of Media in core.

**Review by jbevis — APPROVED** [review]

**Comment by jbevis** [comment]
@jeffsee55 is this PR still relevant? Image field/type appears to be working, and wasn't sure if we still need this one.

**Comment by jeffsee55** [comment]
It is? 🤔 

Do you have it working with all the latest stuff on the starter?

**Comment by jbevis** [comment]
Latest in `new-design` branch of starter is working.

```
...
     {
          type: "image",
          name: "heroImg",
          label: "Hero Image",
        },
...
```

**Pull request closed without merging**